### PR TITLE
Reduce the RAM consumption of RIFE VFI by 2x

### PR DESCRIPTION
**Pull request by gavinliukaiber**
Problem I am trying to fix:
My computer has 32 GB RAM. For a video with 24 fps, 160 second long, 1024 * 768  at float32 resolution, it takes (4*3*1024*576)*(160*24)/1024/1024/1024 = 25.4GB memory. And the original code crashes at line 194 `out = torch.cat(output_frames, dim=0)` because it's trying to allocate the same amount of memory to convert the list of tensor to a single tensor.

Fix: 
+ Avoided torch.cat() by pre-allocating this tensor, and then write directly to this tensor . In this way we can avoid the copying. This reduces RAM consumption by 2x.
+ ~Changed the RIFE's dtype to store the output frames from float32 to float16 (this PR only changes that for VFI, but I am happy to change for all other models as well in this PR if you see fit). This reduces RAM consumption by 2x.~ removed according to comments below.

### CODE REVIEW

**Comment by Fannovel16** [comment]
@gavinliukaiber I pushed an option to use fp16 a while back but most custom nodes assume that IMAGE is full float (https://github.com/Fannovel16/ComfyUI-Frame-Interpolation/issues/48)

**Comment by gavinliukaiber** [comment]
I see, how should we proceed?  But #48 only mentions the upscale node. And  when I linked VFI to VHS_SaveVideo with fp16 it worked as usual.  Separately, the removal of  torch.cat() in this PR  can still save 2x  RAM consumption.

**Comment by Fannovel16** [comment]
I think it would be best to just remove fp16 feature and keeps anything else. Any possible bug with fp16 image tensors would be hard to trace

**Comment by gavinliukaiber** [comment]
Done, please review.

**Comment by Fannovel16** [comment]
LGTM now.
